src/sessions/softphone-session-handler.ts

```diff
@@ -360,12 +363,28 @@ export class SoftphoneSessionHandler extends BaseSessionHandler {
 
     currentEmittedEvent.activeConversationId = this.determineActiveConversationId(session);
 
-    this.log('debug', 'emitting `conversationUpdate`', { event, previousEmittedEvent: this.lastEmittedSdkConversationEvent, currentEmittedEvent, sessionId: session?.id }, { skipServer: true });
+    const previousEventForLogging = this.pruneConversationUpdateForLogging(this.lastEmittedSdkConversationEvent);
+    const currentEventForLogging = this.pruneConversationUpdateForLogging(currentEmittedEvent);
+    this.log('debug', 'emitting `conversationUpdate`', { event, previousEmittedEvent: previousEventForLogging, currentEmittedEvent: currentEventForLogging, sessionId: session?.id }, { skipServer: true });
     this.lastEmittedSdkConversationEvent = currentEmittedEvent;
 
     this.sdk.emit('conversationUpdate', currentEmittedEvent);
   }
 
+  private pruneConversationUpdateForLogging (update: ISdkConversationUpdateEvent): unknown {
```

Comment:
This doesn't work the way you think it does. It's removing the session from the source object. Here's a test that fails though it shouldn't:
```
describe('pruneConversationUpdateForLogging', () => {
  it('should not remove session from original update', () => {
    const lastEmittedSdkConversationEvent = {
      current: [
        { conversationId: 'convo1', session: { id: 'session1' }},
        { conversationId: 'convo2', session: { id: 'session2' }},
      ],
      added: [],
      removed: []
    } as unknown as ISdkConversationUpdateEvent;

    const prunedConvo = handler['pruneConversationUpdateForLogging'](lastEmittedSdkConversationEvent);

    expect(lastEmittedSdkConversationEvent.current[0].session).toBeTruthy();
  });
});
```

Reply:
Thanks for pointing that out. I thought about this a little, but clearly not enough!